Assets/Scripts/UI/InteractArrow.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InteractArrow : MonoBehaviour {
+	[SerializeField] private float bobDistance = 0.5f, bobSpeed = 0.1f;
+	//[SerializeField] private AnimationCurve 
+	private float bobStart, bobCurrent, bobEnd;
+	private bool isIncreasing = false;
+
+	private void Start() {
+		bobCurrent = transform.position.y;
+		bobStart = transform.position.y;
+		bobEnd = transform.position.y + bobDistance;
+	}
+
+	public void BobTheArrow() {
+		if (bobCurrent <= bobStart)
+			isIncreasing = true;
+		else if (bobCurrent >= bobEnd)
+			isIncreasing = false;
+
+		if (isIncreasing) 
+			bobCurrent += bobSpeed * Time.deltaTime;
+		else
+			bobCurrent -= bobSpeed * Time.deltaTime;
```

Comment:
Why are you using a boolean here? Couldn't you just do the following?
```suggestion
		if (bobCurrent <= bobStart)
			bobCurrent += bobSpeed * Time.deltaTime;
		else if (bobCurrent >= bobEnd)
			bobCurrent -= bobSpeed * Time.deltaTime;
```